tests/recovery.test.ts

```diff
@@ -0,0 +1,164 @@
+import { WorkflowContext, OperonWorkflow, OperonTestingRuntime } from "../src/";
+import { generateOperonTestConfig, setupOperonTestDb } from "./helpers";
+import { OperonConfig } from "../src/operon";
+import { OperonTestingRuntimeImpl, createInternalTestRuntime } from "../src/testing/testing_runtime";
+import { OperonWorkflowRecoveryUrl } from "../src/httpServer/server";
+import request from "supertest";
+
+describe("recovery-tests", () => {
+  let config: OperonConfig;
+  let testRuntime: OperonTestingRuntime;
+
+  beforeAll(async () => {
+    config = generateOperonTestConfig();
+    await setupOperonTestDb(config);
+  });
+
+  beforeEach(async () => {
+    testRuntime = await createInternalTestRuntime([LocalRecovery, ExecutorRecovery], config);
+  });
+
+  afterEach(async () => {
+    await testRuntime.destroy();
+  });
+
+  /**
+   * Test for the default local workflow recovery.
+   */
+  class LocalRecovery {
+    static resolve1: () => void;
+    static promise1 = new Promise<void>((resolve) => {
+      LocalRecovery.resolve1 = resolve;
+    });
+
+    static resolve2: () => void;
+    static promise2 = new Promise<void>((resolve) => {
+      LocalRecovery.resolve2 = resolve;
+    });
+
+    static cnt = 0;
+
+    @OperonWorkflow()
+    static async testRecoveryWorkflow(ctxt: WorkflowContext, input: number) {
+      if (ctxt.authenticatedUser === "test_recovery_user" && ctxt.request.url === "test-recovery-url") {
+        LocalRecovery.cnt += input;
+      }
+
+      // Signal the recovery is done.
+      if (LocalRecovery.cnt > input) {
+        LocalRecovery.resolve2();
+      }
+
+      await LocalRecovery.promise1;
+      return ctxt.authenticatedUser;
+    }
+  }
+
+  test("local-recovery", async () => {
```

Comment:
A small comment explaining how this work could be useful. Here my understanding is that we are testing a concurrent workflow execution from a normal `invoke` call and a recovery execution. This test simulate a "local" environment because the `request` parameter does not have a `operon-executorid` header.

Reply:
Yeah exactly.